File: Data Structures & Algorithms/remove-element/submission-3.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int i = 0, j = nums.length - 1;

        while (i <= j) {
            while (j >= 0 && nums[j] == val) {
                j--;
            }
            if (i <= j && nums[i] == val) {
                nums[i] = nums[j];
                j--;
            } else {
                i++;
            }
        }
        return j + 1;
    }
}

File: Data Structures & Algorithms/longest-increasing-subsequence/submission-3.cpp
class Solution {
public:
    int lengthOfLIS(vector<int>& nums) {
        int n = nums.size();
        vector<int> res;

        for (int i=0;i<n;i++){
            if (res.size()==0 || nums[i]>res.back()){
                res.push_back(nums[i]);
                continue;
            }

            if (nums[i]<res.back()){
                int lt = 0;
                while (lt<res.size() && res[lt]<nums[i]){
                    lt++;
                }
                res[lt] = nums[i];
            
            }
           
        }
        return res.size();
    }
};
